### PR TITLE
Add automation for BZ 2168494

### PR DESCRIPTION
**Pull request by shweta83**
### Problem Statement
Automated BZ 2168494

### CODE REVIEW

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_leapp_client.py -k test_leapp_preupgrade_report

File: tests/foreman/ui/test_leapp_client.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    upgrade_path,
+    verify_target_repo_on_satellite,
+    precondition_check_upgrade_and_install_leapp_tool,
+    module_sca_manifest_org,
+):
+    """Test to verify leapp preupgrade report gets generated.
+
+    :id: cbbd38a5-65bc-4491-831a-f1bfeb119d92
+
+    :BZ: 2168494
```
Comment (jyejare) [inline tests/foreman/ui/test_leapp_client.py:45]: Can you also add corresponding Jira card number in the`verifies` token for customer_scenario checks and future features.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7914
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_leapp_client.py -k test_leapp_preupgrade_report --external-logging
Test Result : ================= 1 passed, 15 warnings in 2408.24s (0:40:08) ==================
```

**Comment by shweta83** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_leapp_client.py -k test_leapp_preupgrade_report

File: tests/foreman/ui/test_leapp_client.py (hunk 1 above)
Comment (shweta83) [reply tests/foreman/ui/test_leapp_client.py:45]: Updated.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7916
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_leapp_client.py -k test_leapp_preupgrade_report --external-logging
Test Result : ================= 1 failed, 14 warnings in 2200.27s (0:36:40) ==================
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7918
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_leapp_client.py -k test_leapp_preupgrade_report --external-logging
Test Result : ================= 1 passed, 15 warnings in 2400.64s (0:40:00) ==================
```

File: pytest_fixtures/component/leapp_client.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+from robottelo.config import settings
+from robottelo.constants import PRDS
+from robottelo.hosts import ContentHost
+from robottelo.logging import logger
+
+synced_repos = pytest.StashKey[dict]
+
+RHEL7_VER = '7.9'
+RHEL8_VER = '8.10'
+RHEL9_VER = '9.4'
```
Comment (jyejare) [inline pytest_fixtures/component/leapp_client.py:13]: Use from constants either or define it there.

File: pytest_fixtures/component/leapp_client.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+RHEL_REPOS = {
```
Comment (jyejare) [inline pytest_fixtures/component/leapp_client.py:15]: I think these all should be available in constants or near to repos helper classes, please use from their.

File: pytest_fixtures/component/leapp_client.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+):
+    """Enable and sync RHEL BaseOS, AppStream repositories"""
+    source = upgrade_path['source_version']
+    target = upgrade_path['target_version']
+    all_repos = []
+    for rh_repo_key in RHEL_REPOS:
+        release_version = RHEL_REPOS[rh_repo_key]['releasever']
+        if release_version in str(source) or release_version in target:
+            prod = 'rhel' if 'rhel7' in rh_repo_key else rh_repo_key.split('_')[0]
+            if module_stash[synced_repos].get(rh_repo_key, None):
```
Comment (jyejare) [inline pytest_fixtures/component/leapp_client.py:114]:
```suggestion
            if module_stash[synced_repos].get(rh_repo_key, False):
```

File: tests/foreman/ui/test_leapp_client.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+"""
+
+import pytest
+
+synced_repos = pytest.StashKey[dict]
+
+RHEL7_VER = '7.9'
+RHEL8_VER = '8.10'
+RHEL9_VER = '9.4'
```
Comment (jyejare) [inline tests/foreman/ui/test_leapp_client.py:21]: Same as above.

File: tests/foreman/ui/test_leapp_client.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+:CaseImportance: Critical
+
+:CaseAutomation: Automated
+
+"""
+
+import pytest
+
+synced_repos = pytest.StashKey[dict]
```
Comment (jyejare) [inline tests/foreman/ui/test_leapp_client.py:17]: Doesnt seems its used anywhere in the module.

**Review by jyejare — COMMENTED** [review]
ACK pending minor comments.

File: pytest_fixtures/component/leapp_client.py (hunk 2 above)
Comment (shweta83) [reply pytest_fixtures/component/leapp_client.py:13]: We don't use anywhere else so I kept these locally.

File: pytest_fixtures/component/leapp_client.py (hunk 3 above)
Comment (shweta83) [reply pytest_fixtures/component/leapp_client.py:15]: We don't have any use anywhere other than leapp so I kept in this module only. I feel that the constants common across all the module should be kept in constants file. Wdyt?

File: pytest_fixtures/component/leapp_client.py (hunk 4 above)
Comment (shweta83) [reply pytest_fixtures/component/leapp_client.py:114]: Initially module_stash[synced_repos] is None so we are checking if we don't have cached repository.

File: tests/foreman/ui/test_leapp_client.py (hunk 6 above)
Comment (shweta83) [reply tests/foreman/ui/test_leapp_client.py:17]: I will remove this.

**Review by shubhamsg199 — APPROVED** [review]
Overall looks good

**Review by jyejare — APPROVED** [review]

**Pull request merged**